remoteconfig_test.go

```diff
@@ -569,42 +566,108 @@ func (s *RemoteConfigSuite) TestdownloadJSONValidate() {
 	defer ts.Close()
 
 	c := &SampleConfig{}
-	err := DownloadJSONValidate(ts.URL, c)
+	err := LoadConfigFromURL(ts.URL, c)
 	assert.Nil(s.T(), err)
 }
 
-func (s *RemoteConfigSuite) TestdownloadJSONValidateErrorDownloadFailed() {
+func (s *RemoteConfigSuite) TestLoadConfigFromURL_NotOK() {
 	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		http.Error(w, "JSON Download Failed", http.StatusNotFound)
+		w.WriteHeader(http.StatusNotFound)
+		fmt.Fprintln(w, "Not Found")
 	}))
 	defer ts.Close()
 
+	c := &SQSQueueConfig{}
+	err := LoadConfigFromURL(ts.URL, c)
+	assert.NotNil(s.T(), err)
+	assert.Regexp(s.T(), regexp.MustCompile("Request to 'http://\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}:\\d{1,5}' returned non-200 OK status '404: Not Found'"), err.Error())
```

Comment:
This is fine.

Personally, I find it annoying to change the unit tests when I reword an error message, but I still want to make sure that that I'm hitting code path I'm interested in. I've started creating empty struct types with an Error() method on them and then doing a `s.IsType(ExpectedErrorType{}, err)` to check them. Then I get the knowledge I've hit the correct code path, but I'm free to use whatever error text I want to.